lang/src/accounts/account.rs

```diff
@@ -223,97 +221,141 @@ use std::ops::{Deref, DerefMut};
 /// ```
 /// to access mint accounts.
 #[derive(Clone)]
-pub struct Account<'info, T: AccountSerialize + AccountDeserialize + Owner + Clone> {
```

Comment:
Constricting `T` on `Account`'s functions instead of the type means that certain functions are only available to certain `T`. This allows us to have an `init` function for all `T` that implement `Discriminator` while still allowing non-anchor `T`s (without a discriminator) to be used with `Account`